x-pack/plugins/lens/public/lib/get_all_index_patterns.ts

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  IndexPattern as IndexPatternInstance,
+  IndexPatternsContract,
+} from '../../../../../src/plugins/data/public';
+
+export async function getAllIndexPatterns(
```

Comment:
Nit: Not really a big fan of another kitchen-sink folder for various helpers (we already have a bunch of those). If I understand correctly we will continue to move stuff out of the app component, so at some point this might become an implementation detail of the state management, right?

Maybe we can just put it into `x-pack/plugins/lens/public/utils.ts` for now along with the `getResolvedDateRange` helper?